base/src/main/java/proguard/classfile/util/ConstructorInvocationOffsetFinder.java

```diff
@@ -0,0 +1,107 @@
+package proguard.classfile.util;
+
+import proguard.classfile.*;
+import proguard.classfile.attribute.Attribute;
+import proguard.classfile.attribute.CodeAttribute;
+import proguard.classfile.attribute.visitor.AttributeNameFilter;
+import proguard.classfile.attribute.visitor.AttributeVisitor;
+import proguard.classfile.constant.Constant;
+import proguard.classfile.constant.MethodrefConstant;
+import proguard.classfile.constant.visitor.ConstantVisitor;
+import proguard.classfile.instruction.ConstantInstruction;
+import proguard.classfile.instruction.Instruction;
+import proguard.classfile.instruction.visitor.AllInstructionVisitor;
+import proguard.classfile.instruction.visitor.InstructionOpCodeFilter;
+import proguard.classfile.instruction.visitor.InstructionVisitor;
+import proguard.classfile.visitor.MemberVisitor;
+
+/**
+ * This utility class finds the offset of the invocation to the current or super class constructor
+ * after visiting an <init> method.
+ *
+ * @author Kymeng Tang
+ */
+public class ConstructorInvocationOffsetFinder implements MemberVisitor {
+  private int initOffset = -1;
+
+  public int getConstructorCallOffset() {
+    assert initOffset != -1
+        : "The constructor invocation offset is being requested before visiting any <init> member "
+            + "after instantiation or resetting.";
+    return initOffset;
+  }
+
+  public void reset() {
+    this.initOffset = -1;
+  }
+
+  // MemberVisitor implementation
+  @Override
+  public void visitProgramMethod(ProgramClass programClass, ProgramMethod programMethod) {
+    assert programMethod.getName(programClass).equals(ClassConstants.METHOD_NAME_INIT)
+        : this.getClass().getName()
+            + " only supports constructor but "
+            + programClass.getName()
+            + "."
+            + programMethod.getName(programClass)
```

Comment:
Again, idiomatically this feels like it should be a rejected null visitor instead of asserting it